Ignore Dims of value 1 in Require_Stride_order

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112045
* #111721
* __->__ #111976

Ignore dims of value 1 in require_stride_order since they don't affect layout. Previously, unsqueezed dims would always cause a copy because the stride of 0 would throw off the sorted stride order. This was causing perf problems with require_stride_order in next commit in stack.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

eellison committed:
Ignore Expanded Dims in Require_Stride_order

[ghstack-poisoned]

eellison committed:
Update on "Ignore Expanded 1 Dim in Require_Stride_order"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison committed:
Update on "Ignore Expanded 1 Dim in Require_Stride_order"


Ignore dims of value 1 in require_stride_order since they don't affect layout. Previously, unsqueezed dims would always cause a copy because the stride of 0 would throw off the sorted stride order. This was causing perf problems with require_stride_order in next commit in stack.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Chillee reviewed (approved):
I feel like this is a common pattern - do we not have other utilities for dealing with this?

eellison commented:
There is check_significant_strides in prims_common, but it's a little different than stride_order.

eellison committed:
Update on "Ignore Dims of value 1 in Require_Stride_order"


Ignore dims of value 1 in require_stride_order since they don't affect layout. Previously, unsqueezed dims would always cause a copy because the stride of 0 would throw off the sorted stride order. This was causing perf problems with require_stride_order in next commit in stack.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison committed:
Update on "Ignore Dims of value 1 in Require_Stride_order"


Ignore dims of value 1 in require_stride_order since they don't affect layout. Previously, unsqueezed dims would always cause a copy because the stride of 0 would throw off the sorted stride order. This was causing perf problems with require_stride_order in next commit in stack.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison committed:
Update on "Ignore Dims of value 1 in Require_Stride_order"


Ignore dims of value 1 in require_stride_order since they don't affect layout. Previously, unsqueezed dims would always cause a copy because the stride of 0 would throw off the sorted stride order. This was causing perf problems with require_stride_order in next commit in stack.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison commented:
@pytorchbot merge
